### PR TITLE
refactor: refactor for tasks backport

### PR DESCRIPTION
**Pull request by lesam**
Remove some dead code, and move the FluxLanguageService interface to the fluxlang package.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

**Comment by lesam** [comment]
No changelog for refactors

**Review by danxmoran — APPROVED** [review]

File: authorizer/authorize_find.go [hunk 1]
```diff
@@ -3,6 +3,8 @@ package authorizer
 import (
 	"context"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline authorizer/authorize_find.go:7]: Looks like imports needed cleaned up in general here

File: authorizer/check.go [hunk 2]
```diff
@@ -3,6 +3,8 @@ package authorizer
 import (
 	"context"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline authorizer/check.go:7]: Cleanup imports

File: authorizer/task.go [hunk 3]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline authorizer/task.go:8]: Cleanup imports

File: checks/service.go [hunk 4]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline checks/service.go:8]: Cleanup imports

File: checks/service_external_test.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline checks/service_external_test.go:12]: Cleanup imports

File: checks/service_test.go [hunk 6]
```diff
@@ -4,6 +4,8 @@ import (
 	"context"
 	"testing"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline checks/service_test.go:8]: Cleanup imports

File: cmd/influx/task.go [hunk 7]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline cmd/influx/task.go:9]: Cleanup imports

File: cmd/influx/task_test.go [hunk 8]
```diff
@@ -7,6 +7,8 @@ import (
 	"io/ioutil"
 	"testing"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline cmd/influx/task_test.go:11]: Cleanup imports

File: cmd/influxd/launcher/launcher.go [hunk 9]
```diff
@@ -14,6 +14,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline cmd/influxd/launcher/launcher.go:18]: Cleanup imports

File: cmd/influxd/launcher/launcher_helpers.go [hunk 10]
```diff
@@ -13,6 +13,8 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline cmd/influxd/launcher/launcher_helpers.go:17]: Cleanup imports

File: cmd/influxd/launcher/pkger_test.go [hunk 11]
```diff
@@ -12,6 +12,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline cmd/influxd/launcher/pkger_test.go:16]: Cleanup imports

File: http/api_handler.go [hunk 12]
```diff
@@ -4,8 +4,7 @@ import (
 	"context"
 	"net/http"
 
-	"github.com/influxdata/influxdb/v2/kit/platform"
-	"github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline http/api_handler.go:8]: Cleanup extra whitespace

File: http/check_service.go [hunk 13]
```diff
@@ -9,15 +9,17 @@ import (
 	"path"
 	"time"
 
-	"github.com/influxdata/influxdb/v2/kit/platform"
-	"github.com/influxdata/influxdb/v2/kit/platform/errors"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline http/check_service.go:13]: Extra whitespace

File: http/check_test.go [hunk 14]
```diff
@@ -11,6 +11,8 @@ import (
 	"path"
 	"testing"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline http/check_test.go:15]: Extra whitespace

File: http/notification_rule.go [hunk 15]
```diff
@@ -9,6 +9,8 @@ import (
 	"path"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline http/notification_rule.go:13]: Cleanup imports

File: http/notification_rule_test.go [hunk 16]
```diff
@@ -5,6 +5,8 @@ import (
 	"encoding/json"
 	"testing"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline http/notification_rule_test.go:9]: Cleanup imports

File: http/task_service.go [hunk 17]
```diff
@@ -11,6 +11,8 @@ import (
 	"strconv"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline http/task_service.go:15]: Cleanup imports

File: http/task_service_test.go [hunk 18]
```diff
@@ -13,6 +13,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline http/task_service_test.go:17]: Cleanup imports

File: pkger/service_test.go [hunk 19]
```diff
@@ -13,6 +13,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline pkger/service_test.go:17]: Cleanup imports

File: task/backend/coordinator.go [hunk 20]
```diff
@@ -4,9 +4,10 @@ import (
 	"context"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/coordinator.go:8]: Cleanup imports

File: task/backend/coordinator/coordinator_test.go [hunk 21]
```diff
@@ -6,21 +6,22 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/coordinator/coordinator_test.go:10]: Cleanup imports

File: task/backend/coordinator/support_test.go [hunk 22]
```diff
@@ -3,9 +3,10 @@ package coordinator
 import (
 	"context"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/coordinator/support_test.go:7]: Cleanup imports

File: task/backend/coordinator_test.go [hunk 23]
```diff
@@ -5,10 +5,11 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/coordinator_test.go:9]: Cleanup imports

File: task/backend/executor/executor.go [hunk 24]
```diff
@@ -7,6 +7,8 @@ import (
 	"sync"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/executor/executor.go:11]: Cleanup imports

File: task/backend/executor/executor_metrics.go [hunk 25]
```diff
@@ -3,10 +3,11 @@ package executor
 import (
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/executor/executor_metrics.go:7]: Cleanup imports

File: task/backend/executor/executor_test.go [hunk 26]
```diff
@@ -10,6 +10,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/executor/executor_test.go:14]: Cleanup imports

File: task/backend/executor/limits.go [hunk 27]
```diff
@@ -4,14 +4,16 @@ import (
 	"context"
 	"sort"
 
-	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
+	"github.com/influxdata/influxdb/v2/query/fluxlang"
```
Comment (danxmoran) [inline task/backend/executor/limits.go:9]: Cleanup imports

File: task/backend/executor/limits_test.go [hunk 28]
```diff
@@ -5,13 +5,14 @@ import (
 	"testing"
 	"time"
 
-	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/executor/limits_test.go:9]: Cleanup imports

File: task/backend/middleware/check_middleware.go [hunk 29]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/middleware/check_middleware.go:9]: Cleanup imports

File: task/backend/middleware/middleware.go [hunk 30]
```diff
@@ -5,33 +5,33 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline task/backend/middleware/middleware.go:9]: Cleanup imports

File: task/backend/middleware/middleware_test.go [hunk 31]
```diff
@@ -4,11 +4,11 @@ import (
 	"context"
 	"errors"
 	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline task/backend/middleware/middleware_test.go:7]: Cleanup imports

File: task/backend/middleware/notification_middleware.go [hunk 32]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/middleware/notification_middleware.go:9]: Cleanup imports

File: task/backend/run_recorder.go [hunk 33]
```diff
@@ -6,9 +6,10 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/run_recorder.go:10]: Cleanup imports

File: task/backend/schedulable_task_service.go [hunk 34]
```diff
@@ -5,17 +5,18 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/schedulable_task_service.go:9]: Cleanup imports

File: task/backend/schedulable_task_service_test.go [hunk 35]
```diff
@@ -5,9 +5,10 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/backend/schedulable_task_service_test.go:9]: Cleanup imports

File: task/backend/task.go [hunk 36]
```diff
@@ -4,29 +4,29 @@ import (
 	"context"
 	"time"
 
-	"github.com/influxdata/influxdb/v2/kit/platform"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline task/backend/task.go:8]: Cleanup imports

File: task/mock/executor.go [hunk 37]
```diff
@@ -7,15 +7,16 @@ import (
 	"sync"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/mock/executor.go:11]: Cleanup imports

File: task/mock/task_control_service.go [hunk 38]
```diff
@@ -7,9 +7,10 @@ import (
 	"sync"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/mock/task_control_service.go:11]: Cleanup imports

File: task/servicetest/servicetest.go [hunk 39]
```diff
@@ -16,6 +16,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
+
```
Comment (danxmoran) [inline task/servicetest/servicetest.go:20]: Cleanup imports

File: task/taskmodel/task_test.go [hunk 40]
```diff
@@ -1,18 +1,18 @@
-package influxdb_test
+package taskmodel_test
 
 import (
 	"encoding/json"
+	"github.com/influxdata/influxdb/v2/task/taskmodel"
```
Comment (danxmoran) [inline task/taskmodel/task_test.go:5]: Cleanup imports

**Review by danxmoran — COMMENTED** [review]
Same nitpick applied repeatedly

**Review by danxmoran — APPROVED** [review]

**Pull request merged**